### PR TITLE
E2E test: Netcat

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|---|
|Closes: #2890 |

## Description

In this PR a test has been created to test the Netcat demo case. For this purpose we have created:

- New module: `end_to_end`.
- Common libraries: `event_monitor`.

Each test will include in its directory a `data` folder, where you will find the test configuration as playbooks, and the test itself.

## Configuration options

To launch these tests, we must indicate the path of the inventory:

```
python -m pytest -vvvs tests/end_to_end/test_netcat/ --inventory_path=/home/belen/Repositories/wazuh-qa/tests/end_to_end/general_playbooks/inventory.yml 
```
It should be noted that, for local environments, it will be necessary to create the inventory for the moment. It should contain the following:

```
all:
  hosts:
    wazuh-manager:
      ansible_connection: ssh
      ansible_user: vagrant
      ansible_ssh_private_key_file: <path_to_private_key>
      ansible_python_interpreter: /usr/bin/python3
```

## Checks

- [ ] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [ ] GitHub actions code checks

### CODE REVIEW

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  hosts: wazuh-agent
+  ignore_errors: true
+  tasks:
+    - name: "{{ event_description_netcat }}"
+      command: "{{ command_netcat }}"
+      timeout: 30
+
+    - name: Wait for alert
+      pause:
+        seconds: 5
```
Comment (juliamagan) [inline tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml:19]:
```suggestion
    - name: Wait for alert
      wait_for:
        timeout: 5
```

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: "{{ event_description_kill_netcat }}"
+      become: true
+      command: "{{ command_kill_netcat }}"
```
Comment (juliamagan) [inline tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml:23]: We don't need to pass this command as extra_vars

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but some changes are required.

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml (hunk 1 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml]: Done in a0f66312d81e5ec991fd2d28e1d4d0e92b922162

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml (hunk 2 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml]: Done in a0f66312d81e5ec991fd2d28e1d4d0e92b922162

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml (hunk 1 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml]: Done in 96016ae9ca38441b9392f168996d51e0cf914dc7

File: tests/end_to_end/test_netcat/data/playbooks/configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: Install netcat (CentOS)
+      become: true
+      shell: yum install -y nmap-ncat
+      when: ansible_facts['distribution'] == "CentOS"
+
+    - name: Install netcat (Ubuntu)
+      become: true
+      shell: apt install ncat nmap -y
+      when: ansible_facts['distribution'] == "Ubuntu"
```
Comment (jmv74211) [inline tests/end_to_end/test_netcat/data/playbooks/configuration.yaml:31]: It should be noted in the documentation that this test can only be run on Ubuntu and CentOS.

File: tests/end_to_end/test_netcat/data/playbooks/configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  tasks:
+
+    - name: Configure agent to get a list of running processes
+      become: true
+      blockinfile:
+        path: /var/ossec/etc/ossec.conf
+        insertbefore: </ossec_config>
+        block: |
+          <localfile>
+          <log_format>full_command</log_format>
```
Comment (jmv74211) [inline tests/end_to_end/test_netcat/data/playbooks/configuration.yaml:12]: This can be indented as an xml block?

File: tests/end_to_end/test_netcat/data/playbooks/configuration.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  tasks:
+
+    - name: Configure local rules
+      become: true
+      blockinfile:
+        path: /var/ossec/etc/rules/local_rules.xml
+        insertafter: </groups>
+        block: |
+          <group name="ossec,">
+          <rule id="100050" level="0">
```
Comment (jmv74211) [inline tests/end_to_end/test_netcat/data/playbooks/configuration.yaml:44]: This can be indented as an xml block?

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  hosts: wazuh-manager
+  tasks:
+
+    - name: Truncate file alert.json
+      shell: echo "" > /var/ossec/logs/alerts/alerts.json
+      become: true
+
+- name: Generate events
+  hosts: wazuh-agent
+  ignore_errors: true
```
Comment (jmv74211) [inline tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml:11]: Why is this needed?

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Wait for alert
+      wait_for:
+        timeout: 5
+
+    - name: Kill netcat
+      become: true
+      command: pkill -fx 'nc -l 8000'
+
+- name: Get logs
+  hosts: all
```
Comment (jmv74211) [inline tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml:26]: This is necessary to run in all hosts?

File: tests/end_to_end/test_netcat/data/playbooks/configuration.yaml (hunk 4 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/configuration.yaml:12]: no, linter corrected me that everything should be in the same indentation:

- https://github.com/wazuh/wazuh-qa/runs/6919743996?check_suite_focus=true
- https://github.com/wazuh/wazuh-qa/runs/6919979255?check_suite_focus=true

File: tests/end_to_end/test_netcat/data/playbooks/configuration.yaml (hunk 5 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/configuration.yaml:44]: Same comment as above

File: tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml (hunk 7 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_netcat/data/playbooks/generate_events.yaml]: Done in d50708de21c0bf6e595247a72ad545c1127033e7

**Review by jmv74211 — APPROVED** [review]
